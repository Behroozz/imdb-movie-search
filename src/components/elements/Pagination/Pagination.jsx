import React from 'react'
import { connect } from 'react-redux'
import getItems from '../../../api/getItems'

const Pagination = ({ page, totalPages, term, dispatch }) => {
  console.log(page)
  const nextPage = () => {
    if (page < totalPages) {
      getItems(term, page+= 1, dispatch)
    }
  }
  const prevPage = () => {
    if (page > 1) {
      getItems(term, page-= 1, dispatch)
    }
  }
  return (
    <React.Fragment>
      {totalPages > 0 &&
        <div className="mt-4 ml-4">
          <span className="block w-full text-center md:inline md:w-auto md:text-left">Page {page} of {totalPages}</span>
          <button className="px-4 ml-4 bg-white rounded" onClick={() => prevPage()}>Previous Page</button>
          <button className="px-4 ml-4 bg-white rounded" onClick={() => nextPage()}>Next Page</button>
        </div>
      }
    </React.Fragment>
  )
}

const mapStateToProps = (state) => {
  return {
    page: parseInt(state.data.page),
    totalPages: state.data.total_pages,
    term: state.term
  }
}

export default React.memo(connect(mapStateToProps)(Pagination))