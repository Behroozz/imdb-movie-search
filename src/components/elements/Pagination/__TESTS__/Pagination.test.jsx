import React from 'react'
import renderer from 'react-test-renderer'
import { get } from 'lodash'
import { Pagination } from '../Pagination.jsx'

const mockState = {
  page : 1, 
  totalPages: 3, 
  term: 'water', 
  dispatch: () => {}
}

test('should render a pagination component', () => {
  const component = renderer.create(
    <Pagination 
      page={get(mockState, 'page')} 
      totalPages={get(mockState, 'totalPages')}
      term={get(mockState, 'term')}
      dispatch={get(mockState, 'dispatch')} 
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})