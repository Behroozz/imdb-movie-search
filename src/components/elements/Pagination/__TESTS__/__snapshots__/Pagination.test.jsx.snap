// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a pagination component 1`] = `
<div
  className="mt-4 ml-4"
>
  <span
    className="block w-full text-center md:inline md:w-auto md:text-left"
  >
    Page 
    1
     of 
    3
  </span>
  <button
    className="px-4 ml-4 bg-white rounded"
    onClick={[Function]}
  >
    Previous Page
  </button>
  <button
    className="px-4 ml-4 bg-white rounded"
    onClick={[Function]}
  >
    Next Page
  </button>
</div>
`;
